DigitalLearningSolutions.Web/Services/GroupsService.cs

```diff
@@ -275,26 +299,89 @@ List<Group> groupsForSynchronisation
 
                 foreach (var groupToAddDelegateTo in groupsToAddDelegateTo)
                 {
-                    groupsDataService.AddDelegateToGroup(
-                        delegateId,
-                        groupToAddDelegateTo.GroupId,
-                        clockUtility.UtcNow,
-                        1
-                    );
-
-                    EnrolDelegateOnGroupCourses(
+                    AddDelegateToGroupAndEnrolOnGroupCourses(
                         delegateId,
-                        registrationFieldAnswers.CentreId,
                         accountDetailsData,
                         centreEmail,
-                        groupToAddDelegateTo.GroupId
+                        groupToAddDelegateTo,
+                        registrationFieldAnswers.CentreId
                     );
                 }
             }
 
             transaction.Complete();
         }
 
+        public void SynchroniseJobGroupsOnOtherCentres(
+            int originalDelegateId,
+            int oldJobGroupId,
+            int newJobGroupId,
+            AccountDetailsData accountDetailsData,
+            string? centreEmail
+        )
+        {
+            if (oldJobGroupId == newJobGroupId)
+            {
+                return;
+            }
+
+            var userId = userDataService.GetUserIdFromDelegateId(originalDelegateId);
+            var delegateAccounts = userDataService.GetDelegateAccountsByUserId(userId)
+                .Where(da => da.Id != originalDelegateId);
+
+            foreach (var account in delegateAccounts)
+            {
+                var groupsLinkedToJobGroup = GetGroupsLinkedToJobGroupForCentre(account.CentreId).ToList();
+
+                var oldJobGroupName = jobGroupsDataService.GetJobGroupName(oldJobGroupId);
+                var newJobGroupName = jobGroupsDataService.GetJobGroupName(newJobGroupId);
+
+                var groupToRemoveDelegateFrom = groupsLinkedToJobGroup.Single(
+                    g =>
+                        GroupLabelMatchesAnswer(g.GroupLabel, oldJobGroupName, JobGroupLinkedFieldName)
+                );
+
+                var groupToAddDelegateTo = groupsLinkedToJobGroup.Single(
+                    g =>
+                        GroupLabelMatchesAnswer(g.GroupLabel, newJobGroupName, JobGroupLinkedFieldName)
+                );
+
+                RemoveDelegateFromGroup(account.Id, groupToRemoveDelegateFrom.GroupId);
+
+                AddDelegateToGroupAndEnrolOnGroupCourses(
+                    account.Id,
+                    accountDetailsData,
+                    centreEmail,
+                    groupToAddDelegateTo,
+                    account.CentreId
+                );
+            }
+        }
+
+        private void AddDelegateToGroupAndEnrolOnGroupCourses(

```

Comment:
I find it a bit confusing to have a private and public function with the same name, but with slightly different functionality: 
* calling `AddDelegateToGroup` with different `addedByFieldLink` values
* one having a TransactionScope, the other not

I think ideally, one of these functions would call the other, and they would be one-after-the-other in the file

Reply:
I think it's okay to call the public method `AddDelegateToGroup` because that will always involve enrolling them on courses, and it's natural to assume that GroupsService will deal with the details of what happens when you add a delegate to a group - so we probably don't need to specify at the interface level all the details of what happens.